src/Talis/Persona/Client/Tokens.php

```diff
@@ -539,4 +622,78 @@ protected function personaGetUser($url, $token)
             throw new \Exception("Could not retrieve OAuth response code");
         }
     }
+
+    /**
+     * Create a new user in Persona
+     * @param string $url
+     * @param array $query
+     * @param string $token
+     * @throws \Exception
+     * @return array
+     */
+    protected function personaPostUser($url, $query, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_POST            => true,
+            CURLOPT_URL             => $url,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_POSTFIELDS      => json_encode($query),
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        if (isset($headers['http_code']) && $headers['http_code'] === 200)
+        {
+            return json_decode($response,true);
+        } else
+        {
+            throw new \Exception("Could not retrieve OAuth response code");
+        }
+    }
+
+    /**
+     * Patch a Persona user
+     * @param string $url
+     * @param array $query
+     * @param string $token
+     * @return mixed
+     * @throws \Exception
+     */
+    protected function personaPatchUser($url, $query, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_CUSTOMREQUEST   => 'PUT',
+            CURLOPT_URL             => $url,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_POSTFIELDS      => json_encode($query),
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        $responseDecoded = json_decode($response,true);
+
+        if (isset($headers['http_code']) && $headers['http_code'] === 200)
+        {
+            return $responseDecoded;
+        } else
+        {
+            throw new \Exception($responseDecoded['error_description']);
```

Comment:
are we sure there will always be an error_description? What if there's a network failure during the call for example...